### ISSUE TITLE
Running Tensorflow on GeForce 940M (Ubuntu 14)

### ISSUE THREAD

**Issue by aabramovrepo** [tensorflow/tensorflow#1060]
Hi everyone,

I'm running the CIFAR-10 classification from the Tensorflow for the very first time on my laptop with GeForce 940M. I'm running the training with the pre-defined parameters as follows:

`python cifar10_train.py`

after step 1800 I'm getting the following errors:

`E tensorflow/stream_executor/cuda/cuda_event.cc:33] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
E tensorflow/stream_executor/cuda/cuda_driver.cc:1182] failed to enqueue async memcpy from device to host: CUDA_ERROR_ILLEGAL_ADDRESS; host dst: 0x7ff8e9bf26c0; GPU src: 0x5011c0600; size: 16=0x10
F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:105] Unexpected Event status: 1
I tensorflow/stream_executor/stream.cc:3304] stream 0x35e7190 did not block host until done; was already in an error state
Aborted (core dumped)`

Does anybody have any idea?

By the way, the following fix was required on my side to get the training running:

`from tensorflow.python.platform import gfile
gfile.Exists(...)`

without "ts." in front, it didn't work that way... Any ideas why? I've just forked the repository without any changes.

Thanks a lot in advance for your help! Any advice is kindly appreciated!

**Comment by vrv** [comment]
It looks like the 940M is compute capability 5.0 -- we default to only building for 3.5 and 5.2.

You might want to try building from sources.  See the section [here](https://www.tensorflow.org/versions/v0.6.0/get_started/os_setup.html#installation-for-linux) on installing for Cuda 3.0 to see how you can compile for Cuda 5.0.

@zheng-xq: Is https://github.com/tensorflow/tensorflow/blob/8048088bfb82846078f8023bc6199e6424926624/tensorflow/core/common_runtime/gpu/gpu_device.cc#L683 properly capturing this?  We probably want to just iterate over all compute capabilities and check for equality, unless I'm misunderstanding how compute capabilities work :)

**Comment by aabramovrepo** [comment]
Thanks for your response! Yeah, I will try to build it from sources. By the way, will tensorflow run in future for all models of Nvidia GPUs?

**Comment by vrv** [comment]
Every compute capability you compile for increases the size of the binary significantly, so it is unlikely that we'll support every compute capability Nvidia has ever produced in the default binaries.  But everything >= 3.0 should work if build from sources, although we only test for a small subset.

**Comment by vrv** [comment]
Well, we'll think about it, maybe some people don't care about binary size and want something that works everywhere.  Once we get our release process a bit more solid we could consider this.

**Comment by mznyc** [comment]
Reported similar issue https://github.com/tensorflow/tensorflow/issues/1477
Actually config is for cuda 5.0 when running ./configure    But used cuda7.5 and cuDNN v3 as source to build from.  Is this an issue?

**Comment by girving** [comment]
@vrv: What's the status of this?

**Comment by friesel** [comment]
Having very similar issues as well:
- cifar10 often freezes right after or during local3 (first fc after the two convs)
- Now did this right after or during the first conv:
E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_ADDRESS
F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:198] Unexpected Event status: 1
Aborted (core dumped)
Havent been able to run training for longer than half an hour. Is there a solution for this in the meantime. I understand this might be solved, but was not really planned for. 

Thx

**Comment by friesel** [comment]
Sorry, forgot:
running GeForce 940M with 2 GB VRAM (Acer Aspire E 17)

**Comment by zheng-xq** [comment]
From a different issue #2810, we've found some problems with 940M cuda driver. The problem was solved by: 

https://github.com/tensorflow/tensorflow/issues/2810#issuecomment-232476283

1. Build from source while explicitly setting 5.0 build target in "configure".
2. Or install the latest graphics driver 367.27.

Not sure whether it is related. But it is worth trying.

**Comment by friesel** [comment]
Latest Driver solved it. Thx a bunch

By the way: Scripts running much faster now. Dodnt measure precisely but about a factor 2 on cifar-10

**Comment by suiyuan2009** [comment]
I meet the same problem with GeForce GTX 1080, I build from source and configure compute capability to 6.1. I run a modified version of cifar10, after 9990 step(I set max_step = 10000), program failed.
`2016-08-11 17:26:23.054839: step 9980, loss = 5.73 (55.6 examples/sec; 2.302 sec/batch)

2016-08-11 17:26:47.250385: step 9990, loss = 5.70 (52.9 examples/sec; 2.420 sec/batch)

E tensorflow/stream_executor/cuda/cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_DEINITIALIZED

F tensorflow/core/common_runtime/gpu/gpu_event_mgr.cc:198] Unexpected Event status: 1

Aborted (core dumped)
`

**Comment by gunan** [comment]
@suiyuan2009 did you try the driver update?
Did it resolve the problem for you?

**Comment by suiyuan2009** [comment]
@gunan I solved the problem, but I forget how to solve...

**Comment by gunan** [comment]
OK, then I will close this issue.
